Index: src/pages/trade/ui/Filters.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>import React from 'react'\n\nexport function Filters() {\n  return <div className='flex gap-1.5'>Фильтры</div>\n}\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/pages/trade/ui/Filters.tsx b/src/pages/trade/ui/Filters.tsx
--- a/src/pages/trade/ui/Filters.tsx	(revision 920146f0cb7791f8dfd3a6b20572f953b1e57a50)
+++ b/src/pages/trade/ui/Filters.tsx	(date 1761074830250)
@@ -1,5 +1,39 @@
-import React from 'react'
+import { UISelect, UISelectWithModal, UITabs } from '@/shared'
 
 export function Filters() {
-  return <div className='flex gap-1.5'>Фильтры</div>
+  const tabs = [
+    {
+      value: 'one',
+      label: 'Получено',
+      content: <><h2>Полученое 1</h2></>,
+    },
+    {
+      value: 'second',
+      label: 'Размещено',
+      content: <><h2>Размещено 1</h2></>,
+    },
+  ]
+
+  return (
+    <div className="flex gap-1.5 flex-col">
+      <div className="flex gap-2">
+        <UISelectWithModal>
+
+        </UISelectWithModal>
+
+        <UISelect
+          placeholder="Выберите фрукт"
+          label="Фрукты"
+
+          options={[
+            { value: 'apple', label: 'Apple' },
+            { value: 'banana', label: 'Banana' },
+            { value: 'grapes', label: 'Grapes' },
+          ]}
+        />
+      </div>
+
+      <UITabs tabs={tabs} />
+    </div>
+  )
 }
Index: src/shared/ui/ui-select-with-modal/UISelectWithModal.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/shared/ui/ui-select-with-modal/UISelectWithModal.tsx b/src/shared/ui/ui-select-with-modal/UISelectWithModal.tsx
new file mode 100644
--- /dev/null	(date 1761074830284)
+++ b/src/shared/ui/ui-select-with-modal/UISelectWithModal.tsx	(date 1761074830284)
@@ -0,0 +1,66 @@
+'use client'
+
+import * as React from 'react'
+
+import { cn } from '@/lib/utils'
+import {
+  Drawer,
+  DrawerContent,
+  DrawerHeader,
+  DrawerTitle,
+  DrawerTrigger,
+} from '@/shared/ui/shadcn/drawer'
+import { UIButton } from '@/shared/ui/ui-button'
+
+interface SelectWithModalProps {
+  title?: string // Заголовок в модалке
+  placeholder?: string // Placeholder, если ничего не выбрано
+  value?: string | React.ReactNode // Отображаемое значение в триггере
+  onOpenChange?: (open: boolean) => void // Коллбек при открытии/закрытии
+  children?: React.ReactNode // Контент модалки
+  className?: string
+  disabled?: boolean
+}
+
+export function UISelectWithModal({
+  title,
+  placeholder = 'Select...',
+  value,
+  onOpenChange,
+  children,
+  className,
+  disabled,
+}: SelectWithModalProps) {
+  const [open, setOpen] = React.useState(false)
+
+  const handleOpenChange = (state: boolean) => {
+    setOpen(state)
+    onOpenChange?.(state)
+  }
+
+  return (
+    <Drawer open={open} onOpenChange={handleOpenChange}>
+      <DrawerTrigger asChild>
+        <UIButton
+          variant="outline"
+          className={cn('w-full justify-between', className)}
+          disabled={disabled}
+        >
+          {value || (
+            <span className="text-muted-foreground">{placeholder}</span>
+          )}
+        </UIButton>
+      </DrawerTrigger>
+
+      <DrawerContent className="h-[80vh] rounded-t-2xl p-0 flex flex-col">
+        {title && (
+          <DrawerHeader className="border-b p-4">
+            <DrawerTitle>{title}</DrawerTitle>
+          </DrawerHeader>
+        )}
+
+        <div className="flex-1 overflow-y-auto">{children}</div>
+      </DrawerContent>
+    </Drawer>
+  )
+}
Index: src/shared/ui/ui-select-with-modal/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/shared/ui/ui-select-with-modal/index.ts b/src/shared/ui/ui-select-with-modal/index.ts
new file mode 100644
--- /dev/null	(date 1761074830281)
+++ b/src/shared/ui/ui-select-with-modal/index.ts	(date 1761074830281)
@@ -0,0 +1,1 @@
+export { UISelectWithModal } from './UISelectWithModal'
Index: src/shared/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>export { Button } from './ui/button'\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/shared/index.ts b/src/shared/index.ts
--- a/src/shared/index.ts	(revision 920146f0cb7791f8dfd3a6b20572f953b1e57a50)
+++ b/src/shared/index.ts	(date 1761074837471)
@@ -1,1 +1,5 @@
-export { Button } from './ui/button'
+export { UIButton } from './ui/ui-button'
+export { UIInput } from './ui/ui-input'
+export { UISelect } from './ui/ui-select'
+export { UISelectWithModal } from './ui/ui-select-with-modal'
+export { UITabs } from './ui/ui-tabs'
